### PR TITLE
Prevent imports from .ts files to .tsx files.

### PR DESCRIPTION
**Pull request by seanparsons**
**Problem:**
It was surmised that some of our dependency chains was causing too much to be included in some of the various bundles we produce for the editor.

**Fix:**
Using a slightly cheap trick way of only having `.tsx` files import `.ts` files and not vice-versa, this has created some slight separation in the dependencies, along with also alleviating some of the circular dependencies in the editor.

**Commit Details:**
- Partly addresses #1261.
- Adds a dependency cruiser rule to prevent importing from `.ts` files
  to `.tsx` files, with an exception for `built-in-dependencies.ts`.
- Renames some files in either direction trying to get the rule to be
  satisfied.
- Moved some global mouse position values into a shared file to break
  some dependency chains.
- A few functions and types were moved to prevent the need for some core
  logic to depend on a `.tsx` file.

### CODE REVIEW

**Review by enidemi — APPROVED** [review]

File: editor/src/components/text-utils.ts [hunk 1]
```diff
@@ -247,7 +247,7 @@ export async function measureTextFieldNew(element: ElementInstanceMetadata): Pro
       },
       scale: 1,
     }
-    const textElement = <Text {...textFieldProps} />
+    const textElement = React.createElement(Text, textFieldProps)
```
Comment (Rheeseyb) [inline editor/src/components/text-utils.ts:250]: This one is definitely cheating :eyes:

Comment (seanparsons) [reply editor/src/components/text-utils.ts:250]: This whole function put the fear into me, promise based modifying of the DOM that then triggers actions to be dispatched back into the editor.

Comment (Rheeseyb) [reply editor/src/components/text-utils.ts:250]: Ha ok I just followed this through and see why you've done this! I was going to ask for this to be removed but given how this is used I'd say it's fine to leave as is.

**Review by Rheeseyb — APPROVED** [review]
I really like this. I'd say it's worth disabling the rule for tests though to save us having to deal with renaming .spec files since those don't affect the goal.

**Pull request merged**